shared/actions/chat2/index.js

```diff
@@ -1941,11 +1941,27 @@ const loadCanUserPerform = (state, action) => {
   if (!teamname) {
     return
   }
-  if (!hasCanPerform(state, teamname)) {
+  if (!TeamsConstants.hasCanPerform(state, teamname)) {
     return TeamsGen.createGetTeamOperations({teamname})
   }
 }
 
+// Get the full channel names/descs for a team if we don't already have them.
+function* loadChannelInfos(state, action) {
+  const {conversationIDKey} = action.payload
+  const meta = Constants.getMeta(state, conversationIDKey)
+  const teamname = meta.teamname
+  if (!teamname) {
+    return
+  }
+  if (!TeamsConstants.hasChannelInfos(state, teamname)) {
+    yield Saga.sequentially([
```

Comment:
we don't need this concept in generators themselves, you can just yield the delay then yield the put